services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/QueryOperationTest.java

```diff
@@ -272,6 +272,22 @@ public void generateRequest_consistentRead() {
         assertThat(queryRequest.consistentRead(), is(true));
     }
 
+    @Test
+    public void generateRequest_projectionExpression() {
```

Comment:
Could we test a filter expression together with a projection expression and verify the merging logic is working correctly?